### DISCUSSION TITLE
What's the point of disabling up arrow history when an edit has been made?

### DISCUSSION THREAD

**Discussion by madprops** [fish-shell/fish-shell#9204] [Q&A] [answered]
It happens often that when I use the up arrow to select a previous command, but if I made an edit like a backspace, the history stops working, I have to clear the input and try again. What's the idea behind this?

**Comment by faho** [comment, marked as answer]
The "history" doesn't stop working, the history search is done, and you have now selected your commandline and are editing it.

In the upcoming release you can instead use the new history pager with ctrl-r, which allows you to refine your search term. Here's what that looks like:

![ninja4](https://user-images.githubusercontent.com/5185367/189546019-5bc62551-16d7-491c-ac6b-65d3c46048b4.gif)

**Comment by madprops** [comment]
When is that release coming out? ctrl-r still not working like that in 3.5.1

**Reply by faho** [reply]
This will be in 3.6.0, which is coming out when it's done.
